packages/jest-emotion/src/utils.js

```diff
@@ -26,12 +26,20 @@ function isTagWithClassName(node) {
 }
 
 function getClassNamesFromEnzyme(selectors, node) {
-  // We need to dive if we have selected a styled child from a shallow render
-  const actualComponent = shouldDive(node) ? node.dive() : node
-  // Find the first node with a className prop
-  const components = actualComponent.findWhere(isTagWithClassName)
-  const classes = components.length && components.first().prop('className')
-
+  // We need to get the css prop if we have selected a styled child from a shallow render
+  const isShallow = shouldDive(node)
+  let classes
+  if (isShallow) {
+    // Find the first node with a className prop
```

Comment:
i would extract both "finding" algorithms to separate functions and use them here, then you could rewrite it to smth like this:
```js
function getClassNamesFromEnzyme(selectors, node) {
  const isShallow = shouldDive(node)
  const componentWithClassName = isShallow
    ? findComponentWithClassNameShallow(node)
    : findComponentWithClassName(node)

  if (!componentWithClassName) {
      return /* not sure right not if correct is null, empty string, or empty array, need to check */
  }

  return getClassNames(selectors, componentWithClassName.prop('className'))
}
```